src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -157,16 +157,25 @@ public void createAndLock(Snapshot snapshot, String flightId) {
     logger.debug("createAndLock snapshot " + snapshot.getName());
 
     String sql =
-        "INSERT INTO snapshot (name, description, profile_id, project_resource_id, id, flightid) "
-            + "VALUES (:name, :description, :profile_id, :project_resource_id, :id, :flightid) ";
+        "INSERT INTO snapshot (name, description, profile_id, project_resource_id, id, flightid, mode) "
+            + "VALUES (:name, :description, :profile_id, :project_resource_id, :id, :flightid, :mode) ";
+    var objectMapper = new ObjectMapper();
```

Comment:
Do you need your own `ObjectMapper` for this, or can you use the autowired one? If you do need your own, you should create it as a static member of this class to avoid creating it every time this method is called.